src/components/DateRangePickerInputController.jsx

```diff
@@ -130,14 +133,16 @@ export default class DateRangePickerInputController extends React.Component {
     const {
       startDate,
       isOutsideRange,
+      minimumNights,
       keepOpenOnDateSelect,
       onDatesChange,
     } = this.props;
 
     const endDate = toMomentObject(endDateString, this.getDisplayFormat());
 
     const isEndDateValid = endDate && !isOutsideRange(endDate) &&
-      !isInclusivelyBeforeDay(endDate, startDate);
+      (!isInclusivelyBeforeDay(endDate, startDate) ||
+      (minimumNights === 0 && isSameDay(endDate, startDate)));
```

Comment:
This is a bit of a nit, but could we change this to:
```
(
  !isInclusivelyBeforeDay(endDate, startDate) ||
  (minimumNights === 0 && isSameDay(endDate, startDate))
);
```

Also, I feel like it's a bit weird that we don't check for minimum nights requirements in other scenarios. Maybe this should just be:
```
 isInclusivelyAfterDay(endDate, startDate + minimumNights)
```
Would that work?